src/cropper.sol

```diff
@@ -0,0 +1,495 @@
+        require((z = x + y) >= x);
+    }
+    function sub(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require((z = x - y) <= x);
+    }
+    function mul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require(y == 0 || (z = x * y) / y == x);
+    }
+    function wmul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = mul(x, y) / WAD;
+    }
+    function rmul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = mul(x, y) / RAY;
+    }
+    function rdiv(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = mul(x, RAY) / y;
+    }
+
+    // --- Auction ---
+
+    // get the price directly from the OSM
+    // Could get this from rmul(Vat.ilks(ilk).spot, Spotter.mat()) instead, but
+    // if mat has changed since the last poke, the resulting value will be
+    // incorrect.
+    function getFeedPrice() internal returns (uint256 feedPrice) {
+        (PipLike pip, ) = spotter.ilks(ilk);
+        (bytes32 val, bool has) = pip.peek();
+        require(has, "CropClipper/invalid-price");
+        feedPrice = rdiv(mul(uint256(val), BLN), spotter.par());
+    }
+
+    // start an auction
+    // note: trusts the caller to transfer collateral to the contract
+    // The starting price `top` is obtained as follows:
+    //
+    //     top = val * buf / par
+    //
+    // Where `val` is the collateral's unitary value in USD, `buf` is a
+    // multiplicative factor to increase the starting price, and `par` is a
+    // reference per DAI.
+    function kick(
+        uint256 tab,  // Debt                   [rad]
+        uint256 lot,  // Collateral             [wad]
+        address usr,  // Address that will receive any leftover collateral; additionally assumed here to be the liquidated Vault.
+        address kpr   // Address that will receive incentives
+    ) external auth lock isStopped(1) returns (uint256 id) {
+        // Input validation
+        require(tab  >          0, "CropClipper/zero-tab");
+        require(lot  >          0, "CropClipper/zero-lot");
+        require(usr != address(0), "CropClipper/zero-usr");
+        id = ++kicks;
+        require(id   >          0, "CropClipper/overflow");
+
+        active.push(id);
+
+        sales[id].pos = active.length - 1;
+
+        sales[id].tab = tab;
+        sales[id].lot = lot;
+        sales[id].usr = usr;
+        sales[id].tic = uint96(block.timestamp);
+
+        uint256 top;
+        top = rmul(getFeedPrice(), buf);
+        require(top > 0, "CropClipper/zero-top-price");
+        sales[id].top = top;
+
+        // incentive to kick auction
+        uint256 _tip  = tip;
+        uint256 _chip = chip;
+        uint256 coin;
+        if (_tip > 0 || _chip > 0) {
+            coin = add(_tip, wmul(tab, _chip));
+            vat.suck(vow, kpr, coin);
+        }
+
+        // TODO: harvest rewards for user once a reverse UrnProxy mapping is implemented.
```

Comment:
Is there a plan to implement this? You can do it here by reading the user from the urn proxy. IMO it is the correct user experience.